tests/e2e/utils/DriverHelper.ts

```diff
@@ -102,19 +102,42 @@ export class DriverHelper {
                     Logger.error(`DriverHelper.waitVisibility - failed with exception, out of attempts - ${err}`);
                     throw err;
                 }
-                continue;
+
+                if (err instanceof error.TimeoutError) {
+                    if (attempts !== 1) { // waitVisibility was spamming other methods when the number of attempts was 1 - only show message if attempts > 1
+                        Logger.trace(`DriverHelper.waitVisibility - Polling timed out attempt #${(i + 1)}, retrying with ${polling}ms timeout`);
+                    }
+                    continue;
+                }
+
+                if (err instanceof error.NoSuchWindowError) { // sometimes waitVisibility fails with NoSuchWindowError when the check is run too soon before the page loads
```

Comment:
Added logic for capturing NoSuchWindowError when waitVisibility is executed before page is loaded.